hack/build-base-images.sh

```diff
@@ -5,10 +5,29 @@
 STARTTIME=$(date +%s)
 source "$(dirname "${BASH_SOURCE}")/lib/init.sh"
 
+HOST_ARCH=$(os::build::host_arch)
+
+DOCKERFILE="Dockerfile"
+
+if [[ "${HOST_ARCH}" != "amd64" ]]; then
+  DOCKERFILE+=".${HOST_ARCH}"
+fi
+
 # determine the correct tag prefix
 tag_prefix="${OS_IMAGE_PREFIX:-"openshift/origin"}"
 
+BASE_EXTRA_TAG=""
+if [[ "${HOST_ARCH}" == "amd64" ]]; then
+  BASE_EXTRA_TAG="${tag_prefix}-base"
+fi
+
 # Build the base image without the default image args
-OS_BUILD_IMAGE_ARGS="${OS_BUILD_IMAGE_BASE_ARGS-}" os::build::image "${OS_ROOT}/images/base" "${tag_prefix}-base"
+OS_BUILD_IMAGE_ARGS="${OS_BUILD_IMAGE_BASE_ARGS-}" os::build::image "${OS_ROOT}/images/base" "${tag_prefix}-base-${HOST_ARCH}" ${DOCKERFILE} ${BASE_EXTRA_TAG}
```

Comment:
We didn't need to do this before (for `centos7` and `rhel7`) by curating the images that are available at build-time on the local daemon. Why can't that approach work for this?